tests/registry_test.go

```diff
@@ -0,0 +1,166 @@
+package tests
+
+import (
+	"time"
+
+	"github.com/deis/workflow-e2e/tests/cmd"
+	"github.com/deis/workflow-e2e/tests/cmd/apps"
+	"github.com/deis/workflow-e2e/tests/cmd/auth"
+	"github.com/deis/workflow-e2e/tests/model"
+	"github.com/deis/workflow-e2e/tests/settings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("deis registry", func() {
+
+	Context("with an existing user", func() {
+
+		var user model.User
+
+		BeforeEach(func() {
+			user = auth.Register()
+		})
+
+		AfterEach(func() {
+			auth.Cancel(user)
+		})
+
+		Context("who owns an existing app", func() {
+
+			var app model.App
+
+			BeforeEach(func() {
+				app = apps.Create(user, "--no-remote")
+			})
+
+			AfterEach(func() {
+				apps.Destroy(user, app)
+			})
+
+			Specify("that user can list that app's registry information", func() {
+				sess, err := cmd.Start("deis registry:list --app=%s", &user, app.Name)
+				Eventually(sess).Should(Say("=== %s Registry", app.Name))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+			})
+
+			Specify("that user cannot unset an invalid registry information", func() {
+				sess, err := cmd.Start("deis registry:unset --app=%s munkafolyamat", &user, app.Name)
+				// TODO: should unsetting a bogus registry info return 0 (success?)
```

Comment:
That's a copy from the tags test, actually :) So blame whoever wrote the tags test ;)